libbpf-tools/fsdist.c

```diff
@@ -0,0 +1,489 @@
+		    && !fentry_exists(fn_name, module)) {
+			support_fentry = false;
+			break;
+		}
+	}
+	return support_fentry;
+}
+
+static void fentry_set_attach_target(struct fsdist_bpf *obj)
+{
+	bpf_program__set_attach_target(obj->progs.file_read_fentry, 0,
+		fs_configs[fs_type].op_funcs[READ]);
+	bpf_program__set_attach_target(obj->progs.file_read_fexit, 0,
+		fs_configs[fs_type].op_funcs[READ]);
+	bpf_program__set_attach_target(obj->progs.file_write_fentry, 0,
+		fs_configs[fs_type].op_funcs[WRITE]);
+	bpf_program__set_attach_target(obj->progs.file_write_fexit, 0,
+		fs_configs[fs_type].op_funcs[WRITE]);
+	bpf_program__set_attach_target(obj->progs.file_open_fentry, 0,
+		fs_configs[fs_type].op_funcs[OPEN]);
+	bpf_program__set_attach_target(obj->progs.file_open_fexit, 0,
+		fs_configs[fs_type].op_funcs[OPEN]);
+	bpf_program__set_attach_target(obj->progs.file_sync_fentry, 0,
+		fs_configs[fs_type].op_funcs[FSYNC]);
+	bpf_program__set_attach_target(obj->progs.file_sync_fexit, 0,
+		fs_configs[fs_type].op_funcs[FSYNC]);
+	if (fs_configs[fs_type].op_funcs[GETATTR]) {
+		bpf_program__set_attach_target(obj->progs.getattr_fentry, 0,
+		fs_configs[fs_type].op_funcs[GETATTR]);
+		bpf_program__set_attach_target(obj->progs.getattr_fexit, 0,
+			fs_configs[fs_type].op_funcs[GETATTR]);
+	} else {
+		bpf_program__set_autoload(obj->progs.getattr_fentry, false);
+		bpf_program__set_autoload(obj->progs.getattr_fexit, false);
+	}
+}
+
+static void disable_fentry(struct fsdist_bpf *obj)
+{
+	bpf_program__set_autoload(obj->progs.file_read_fentry, false);
+	bpf_program__set_autoload(obj->progs.file_read_fexit, false);
+	bpf_program__set_autoload(obj->progs.file_write_fentry, false);
+	bpf_program__set_autoload(obj->progs.file_write_fexit, false);
+	bpf_program__set_autoload(obj->progs.file_open_fentry, false);
+	bpf_program__set_autoload(obj->progs.file_open_fexit, false);
+	bpf_program__set_autoload(obj->progs.file_sync_fentry, false);
+	bpf_program__set_autoload(obj->progs.file_sync_fexit, false);
+	bpf_program__set_autoload(obj->progs.getattr_fentry, false);
+	bpf_program__set_autoload(obj->progs.getattr_fexit, false);
+}
+
+static void disable_kprobes(struct fsdist_bpf *obj)
+{
+	bpf_program__set_autoload(obj->progs.file_read_entry, false);
+	bpf_program__set_autoload(obj->progs.file_read_exit, false);
+	bpf_program__set_autoload(obj->progs.file_write_entry, false);
+	bpf_program__set_autoload(obj->progs.file_write_exit, false);
+	bpf_program__set_autoload(obj->progs.file_open_entry, false);
+	bpf_program__set_autoload(obj->progs.file_open_exit, false);
+	bpf_program__set_autoload(obj->progs.file_sync_entry, false);
+	bpf_program__set_autoload(obj->progs.file_sync_exit, false);
+	bpf_program__set_autoload(obj->progs.getattr_entry, false);
+	bpf_program__set_autoload(obj->progs.getattr_exit, false);
+}
+
+static int attach_kprobes(struct fsdist_bpf *obj)
+{
+	long err;
+
+	// READ
+	obj->links.file_read_entry = bpf_program__attach_kprobe(
```

Comment:
try to keep them on a single line as well, it's way more readable. It's fine to have a bit longer lines, if necessary